packages/runner/src/runner.ts

```diff
@@ -16,6 +16,68 @@ type MaybeArray<T> = T | T[];
 
 const isPrimitive = (obj: any) => ["string", "number", "boolean"].includes(typeof obj);
 
+
+export function loopStream<T extends unknown>(
+    stream: Readable,
+    iter: (chunk: Buffer) => { action: "continue" } | { action: "end", data: T, unconsumedData?: Buffer }
+): Promise<T> {
+    return new Promise((res, rej) => {
+        const onReadable = () => {
+            let chunk;
+
+            while ((chunk = stream.read()) !== null) {
+                const result = iter(chunk);
+
+                if (result.action === "continue") {
+                    continue;
+                }
+
+                stream.off("error", rej);
+                stream.off("readable", onReadable);
+                if (result.unconsumedData?.length) {
+                    stream.unshift(result.unconsumedData);
+                }
+
+                res(result.data);
+                break;
+            }
+        };
+
+        stream.on("error", rej);
+        stream.on("readable", onReadable);
```

Comment:
You should execute the `onReadable` here once during loopStream call since the stream may already be readable. It's safe, as `read` will not yield any results otherwise and the body of the `while` will not be executed.

Reply:
fixed